features/admin.actions.v1/models/actions.ts

```diff
@@ -0,0 +1,343 @@
+     */
+    uri: string;
+    /**
+     * Authentication configurations of the Action.
+     */
+    authentication: AuthenticationInterface;
+}
+
+/**
+ *  Endpoint authentication configuration.
+ */
+export interface AuthenticationInterface {
+    /**
+     * Authentication Type.
+     */
+    type: AuthenticationType;
+    /**
+     * Authentication properties.
+     */
+    properties: AuthPropertiesInterface;
+}
+
+/**
+ * Authentication Properties.
+ */
+export interface AuthPropertiesInterface {
+    /**
+     * Username auth property.
+     */
+    username?: string;
+    /**
+     * Password auth property.
+     */
+    password?: string;
+    /**
+     * Access Token auth property.
+     */
+    accessToken?: string;
+    /**
+     * Header auth property.
+     */
+    header?: string;
+    /**
+     * Value auth property.
+     */
+    value?: string;
+}
+
+/**
+ *  API Key Authentication Properties.
+ */
+export interface ApiKeyAuthPropertiesInterface {
+    /**
+     * Header auth property.
+     */
+    header: string;
+    /**
+     * Value auth property.
+     */
+    value: string;
+}
+
+/**
+ *  Action Basic Response.
+ */
+export interface ActionBasicResponseInterface {
+    /**
+     * ID of the Action.
+     */
+    id?: string;
+    /**
+     * Type of the Action.
+     */
+    type?: ActionType;
+    /**
+     * Name of the Action.
+     */
+    name?: string;
+    /**
+     * Description of the Action.
+     */
+    description?: string;
+    /**
+     * Status of the Action.
+     */
+    status?: ActionStatus;
+}
+
+/**
+ *  Action Response.
+ */
+export interface ActionResponseInterface {
+    /**
+     * ID of the Action.
+     */
+    id?: string;
+    /**
+     * Type of the Action.
+     */
+    type?: ActionType;
+    /**
+     * Name of the Action.
+     */
+    name?: string;
+    /**
+     * Description of the Action.
+     */
+    description?: string;
+    /**
+     * Status of the Action.
+     */
+    status?: ActionStatus;
+    /**
+     * Endpoint configuration of the Action.
+     */
+    endpoint: EndpointResponseInterface;
+}
+
+/**
+ *  Endpoint Response.
+ */
+export interface EndpointResponseInterface {
+    /**
+     * External endpoint.
+     */
+    uri: string;
+    /**
+     * Authentication configurations of the Action.
+     */
+    authentication: AuthenticationResponseInterface;
+}
+
+/**
+ *  Endpoint authentication response.
+ */
+export interface AuthenticationResponseInterface {
+    /**
+     * Authentication Type.
+     */
+    type: AuthenticationType;
+}
+
+/**
+ *  Action Update configuration.
+ */
+export interface ActionUpdateInterface {
+    /**
+     * Name of the Action.
+     */
+    name?: string;
+    /**
+     * Description of the Action.
+     */
+    description?: string;
+    /**
+     * Endpoint configuration of the Action.
+     */
+    endpoint?: EndpointUpdateInterface;
+}
+
+/**
+ *  Endpoint update configuration.
+ */
+export interface EndpointUpdateInterface {
```

Comment:
use Partial from the endpoint interface

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/6722/commits/d96314b591f6f6f33b709b0600c399b789e7fcef